src/components/Comments/commentsSlicer.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-cycle
+import { RootState } from '../../app/store';
+import { Comment, CommentData } from '../../types/Comment';
+import { Post } from '../../types/Post';
+// eslint-disable-next-line import/no-cycle
+import { fetchUserPosts } from '../Posts/userPostsSlicer';
+import * as commentsApi from '../../api/comments';
+import { ErrorTypes, LoadingStatus } from '../../types/enums';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface CommentsState {
+  comments: Comment[];
+  loading: LoadingStatus;
+  error: string,
+  selectedPost: Post | null,
+}
+
+const initialState: CommentsState = {
+  comments: [],
+  loading: LoadingStatus.Idle,
+  error: '',
+  selectedPost: null,
+};
+
+export const fetchPostComments = createAsyncThunk(
+  'comments/fetchComments',
+  async (post: Post) => {
+    return {
+      selectedPost: post,
+      comments: await commentsApi.getPostComments(post.id),
+    };
+  },
+);
+
+export const deleteComment = createAsyncThunk(
+  'comments/deleteComment',
+  async (commentId: number) => {
+    await commentsApi.deleteComment(commentId);
+
+    return { id: commentId };
+  },
+);
+
+export const createComment
+  = createAsyncThunk<Comment, CommentData, { state: RootState }>(
+    'comments/createComment',
+    async ({ name, email, body }: CommentData, { getState }) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I guess it's new comment and they all need to be recorded. This affects the cleanliness of the code and the general understanding of what is happening in it. Try to fix and  fill free to ask in fe_chat if you face with any problems